xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -230,7 +235,8 @@ protected Subchannel delegate() {
   final class WeightedRoundRobinPicker extends ReadyPicker {
     private final List<Subchannel> list;
     private volatile EdfScheduler scheduler;
-    private volatile boolean rrMode;
+    @VisibleForTesting
+    volatile boolean rrMode;
```

Comment:
Can rrMode be replaced with something like bool isRrMode() { this.scheduler == null}? If so, I think it will less error-prone.

When the picker is created, it's either "rrMode && no scheduler" or "not rrMode and with scheduler". Later when rrMode = true is set, the current scheduler won't get called until replaced with a new one but we always set rrMode = false at the same time. So I think we just need to check if scheduler is null?

Reply:
yea i think it is a good idea to remove rrMode boolean and use scheduler to indicate using round robin instead.
Since scheduler is volatile, and the picker and scheduler updater can be from different thread, we need to make a copy when do the checking.